frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -194,6 +221,114 @@ const updateItemAppConnectTo = (
   return k8sPatch(model, item, patch);
 };
 
+export const checkServiceBindingPermissions = async (
+  namespace: string,
+  action: K8sVerb,
+  name?: string,
+) => {
+  const access = await checkAccess({
+    group: ServiceBindingRequestModel.apiGroup,
+    resource: ServiceBindingRequestModel.plural,
+    verb: action,
+    name,
+    namespace,
+  });
+
+  return access.status.allowed;
+};
+
+export const createServiceBinding = (
+  source: K8sResourceKind,
+  target: K8sResourceKind,
+): Promise<any> => {
+  if (!source || !target || source === target) {
+    return Promise.reject();
+  }
+
+  const targetName = _.get(target, 'metadata.name');
+  const sourceName = _.get(source, 'metadata.name');
+  const namespace = _.get(source, 'metadata.namespace');
+  const targetResourceGroup = _.split(_.get(target, 'metadata.ownerReferences[0].apiVersion'), '/');
+  const targetResourceKind = _.get(target, 'metadata.ownerReferences[0].kind');
+  const targetResourceRefName = _.get(target, 'metadata.ownerReferences[0].name');
+  const sbrName = `${sourceName}-${targetName}-sbr`;
+
+  const serviceBindingRequest = {
+    apiVersion: 'apps.openshift.io/v1alpha1',
+    kind: 'ServiceBindingRequest',
+    metadata: {
+      name: sbrName,
+      namespace,
+    },
+    spec: {
+      applicationSelector: {
+        resourceRef: sourceName,
+        group: 'apps.openshift.io',
+        version: 'v1',
+        resource: modelFor(source.kind).plural,
+      },
+      backingServiceSelector: {
+        group: targetResourceGroup[0],
+        version: targetResourceGroup[1],
+        kind: targetResourceKind,
+        resourceRef: targetResourceRefName,
+      },
+    },
+  };
+
+  return checkServiceBindingPermissions(namespace, 'create')
+    .then((isAllowed) => {
+      if (isAllowed) {
+        return k8sCreate(ServiceBindingRequestModel, serviceBindingRequest);
+      }
+      const err: Error = {
+        name: 'Service Binding Creation',
+        message: 'You do not have permission to create this service binding',
+      };
+      throw err;
+    })
+    .catch((err) => {
+      throw err;
+    });
+};
+
+export const removeServiceBinding = (
+  source: K8sResourceKind,
+  target: K8sResourceKind,
+): Promise<any> => {
+  if (!source || !target || source === target) {
+    return Promise.reject();
+  }
+  const targetName = _.get(target, 'metadata.name');
+  const sourceName = _.get(source, 'metadata.name');
+  const namespace = _.get(source, 'metadata.namespace');
+  const sbrName = `${sourceName}-${targetName}-sbr`;
+
+  const serviceBindingRequest = {
+    apiVersion: 'apps.openshift.io/v1alpha1',
+    kind: 'ServiceBindingRequest',
+    metadata: {
+      name: sbrName,
+      namespace,
+    },
+  };
+
+  return checkServiceBindingPermissions(namespace, 'delete', sbrName)
```

Comment:
The check here isn't really useful because the operation would have failed anyways if they didn't have permission. We should not be allowing the user to perform the operation in the first place. I say remove the rbac checks here for now. When we move to the new topology it'll be easier to add rbac to prevent the user from seeing the delete icon in the first place.